### PR TITLE
CE support

### PR DESCRIPTION
**Pull request by sklencar**
Changes due to support for CE Mergin server. The idea is to use primary mergin core endpoints. Request related to subscription are sent extra. Therefore login and userInfo request were changed.

Note that there is a possibility to corrupt data by mixing projects from different servers ATM: https://github.com/lutraconsulting/input/issues/1347

**Squash and Merge** option can be ideally used.

Account page with server supporting subscription:
Screenshot 2021-04-22 at 10 06 06

Account page with server NOT supporting subscription (looks quite empty):
Screenshot 2021-04-22 at 10 08 58

related to #1289

### CODE REVIEW

**Comment by PeterPetrik** [comment]
nice, Viktor, can you please

1. add screenshot WHEN billing is on and when is OFF
2. add a manual-input test against CE mergin server?

File: app/qml/ProjectPanel.qml [hunk 1]
```diff
@@ -434,11 +436,9 @@ Item {
           if (__merginApi.apiVersionStatus === MerginApiStatus.OK && stackView.currentItem.objectName === "authPanel") {
             if (__merginApi.userAuth.hasAuthData()) {
               refreshProjectList()
-            } else if (toolbar.highlighted !== homeBtn.text) {
-              if (stackView.currentItem.objectName !== "authPanel") {
+            } else if (stackView.currentItem.objectName !== "authPanel") {
                 stackView.push(authPanelComp, {state: "login"})
               }
-            }
```
Comment (PeterPetrik) [inline app/qml/ProjectPanel.qml:441]: what is this change related to?

File: core/merginapi.cpp [hunk 2]
```diff
@@ -663,6 +664,28 @@ void MerginApi::getUserInfo( )
   connect( reply, &QNetworkReply::finished, this, &MerginApi::getUserInfoFinished );
 }
 
+void MerginApi::getSubscriptionInfo()
+{
+  if ( !apiSupportsSubscriptions() )
+  {
+    CoreUtils::log( "subscription info", QStringLiteral( "Request skipped - server doesn't support subscriptions." ) );
```
Comment (PeterPetrik) [inline core/merginapi.cpp:671]: this should be just qDebug() not sure why it needs to be in the user logs

File: core/merginapi.cpp [hunk 3]
```diff
@@ -823,7 +846,8 @@ void MerginApi::authorizeFinished()
     {
       QJsonObject docObj = doc.object();
       mUserAuth->setFromJson( docObj );
-      mUserInfo->setFromJson( docObj );
+
+      getUserInfo();
```
Comment (PeterPetrik) [inline core/merginapi.cpp:850]: not sure about his... user info is not part of /login endpoint?

File: core/merginapi.cpp [hunk 4]
```diff
@@ -2120,6 +2144,34 @@ void MerginApi::getUserInfoFinished()
...
+  Q_ASSERT( r );
+
+  if ( r->error() == QNetworkReply::NoError )
+  {
+    CoreUtils::log( "subscription info", QStringLiteral( "Success" ) );
+    QJsonDocument doc = QJsonDocument::fromJson( r->readAll() );
+    if ( doc.isObject() )
+    {
+      QJsonObject docObj = doc.object();
+      mUserInfo->setSubscriptionInfoFromJson( docObj );
```
Comment (PeterPetrik) [inline core/merginapi.cpp:2159]: hmm, this is not good. we should have CLEAR separatarion what is get from which endpoint. so what you get from 
login endpoint should be 1 or 2 structs, what you get from subscripion endpoint should be separate struct. ideally named mUserSubscriptionInfo

File: app/qml/ProjectPanel.qml (hunk 1 above)
Comment (sklencar) [reply app/qml/ProjectPanel.qml:441]: This is just fixed leftover from refactoring - `toolbar` component doesn't exists. I will it move to a separate PR.

**Review by PeterPetrik — APPROVED** [review]

**Comment by sklencar** [comment]
closes #1289
